pkg/controller/telemetry/metric.go

```diff
@@ -163,37 +171,50 @@ func (m *MetricController) Run(ctx context.Context, mapOfTcpInfo *ebpf.Map) {
 		}
 	}()
 
-	// Register metrics to Prometheus and start Prometheus server
-	go RunPrometheusClient(ctx)
-
-	rec := ringbuf.Record{}
-	data := requestMetric{}
-
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		default:
+			rec := ringbuf.Record{}
 			if err := reader.ReadInto(&rec); err != nil {
 				log.Errorf("ringbuf reader FAILED to read, err: %v", err)
-				continue
+				return
```

Comment:
why did it change from `continue` to `return`? 

`getConnectDataFromRingBuf` run as goroutine and don't handle errors. Event if it fails, only the goroutine exits and the rest of the system is completely unaware. 

We need a fault-tolerant mechanism or an overall exit after failure